### DISCUSSION TITLE
Lexical (out of beta)

### DISCUSSION THREAD

**Discussion by zubricks** [payloadcms/payload#4201] [Roadmap]
The following needs to be done in order to bring Lexical RTE out of beta into a more stable state.

- Close out all open issues → [[https://github.com/payloadcms/payload/issues?q=is:issue+is:open+label:"plugin:+richtext-lexical"](https://github.com/payloadcms/payload/issues?q=is:issue+is:open+label:%22plugin:+richtext-lexical%22)](https://github.com/payloadcms/payload/issues?q=is:issue+is:open+label:%22plugin:+richtext-lexical%22)
- Finalize the lexical config interface
- Implement some ToDos, like missing validations for some nodes, etc.

EDIT: Lexical is officially out of beta in **[3.0](https://github.com/payloadcms/payload/releases/tag/v3.0.0)**

**Comment by vibhor-web-headout** [comment]
Hey Team,

Hope you're doing well! I'm looking to move some custom components from Slate to Lexical and have a couple of quick questions:

1. Any idea when we might wrap up the above task?

2. With the upcoming changes in Lexical's config interface, do you think it'll mean some tweaks (big/small/none) for the components already moved from Slate to `payload/rich-text-lexical`?

Thanks a bunch!

**Reply by AlessioGr** [reply]
> Any idea when we might wrap up the above task?

Converting data from Slate and Lexical is easy. See: https://payloadcms.com/docs/rich-text/lexical#migration-via-migration-script

The main complexity comes from the amount of different richtext fields and how nested they are - because it will complicate your migration script.

As for moving the custom component implementation itself from Slate or Lexical, it really depends on the component. Lexical is very different from Slate, and you'll have to learn it first.

> With the upcoming changes in Lexical's config interface, do you think it'll mean some tweaks (big/small/none) for the components already moved from Slate to payload/rich-text-lexical?

Yep! There is one update with big breaking changes in the interface coming soon. Probably this month, but I don't have an ETA for sure. For any future updates after that, we want to keep the breaking changes low, which is why we aim to combine them in that one, big update.
